conformance/tests/httproute-request-mirror.go

```diff
@@ -58,6 +58,28 @@ var HTTPRouteRequestMirror = suite.ConformanceTest{
 				Backend:    "infra-backend-v1",
 				MirroredTo: "infra-backend-v2",
 				Namespace:  ns,
+			}, {
+				Request: http.Request{
+					Path: "/mirror-and-modify-headers",
+					Headers: map[string]string{
+						"X-Header-Remove":     "remove-val",
+						"X-Header-Add-Append": "append-val-1",
+					},
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/mirror-and-modify-headers",
+						Headers: map[string]string{
+							"X-Header-Add":        "header-val-1",
+							"X-Header-Add-Append": "append-val-1,header-val-2",
+							"X-Header-Set":        "set-overwrites-values",
+						},
+					},
+					AbsentHeaders: []string{"X-Header-Remove"},
+				},
```

Comment:
As far as I can tell, this doesn't actually verify that the request to the mirrored backend received the modified headers (https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/http/mirror.go#L32). Assuming my understanding is correct, we should probably either add a comment here or create an issue to track improving that. @LiorLieberman would probably know best.

Reply:
Thanks both for pointing this out, I created a separate issue to improve this later https://github.com/kubernetes-sigs/gateway-api/issues/2142